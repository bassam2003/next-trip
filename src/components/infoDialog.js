import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { Dialog, DialogContent } from '@material-ui/core'
import { header } from '../utils/styleConstants'

const styles = {
  header,
  listing: {
    marginBottom: '10px'
  },
  content: {
    textAlign: 'center',
    padding: '10px',
  }
}

export const InfoDialog = ({ 
  currentStopText = '', 
  open = false, 
  departures = [],
  handleClose, 
  classes = {},
  currentRouteData = {},
  currentDirectionData = {}, 
}) => {
  return(
    <Dialog open={open} fullWidth onClose={handleClose} >
      <div className={classes.header}>
        <strong>Route:</strong> {currentRouteData.Description}<br/>
        <strong>Stop:</strong> {currentStopText}<br/>
        <strong>Direction:</strong> {currentDirectionData.Text} 
      </div>
      <DialogContent className={classes.content}>
        {departures.length === 0 ? 
          <h4>No Departures left today</h4>:
          departures.map(departure => 
            <div className={classes.listing}>
              <span>{departure.Description}</span><br/>
              <span>Departs At: {departure.DepartureText}</span>
            </div>
          )
        }
      </DialogContent>
    </Dialog>
  )
}

export default withStyles(styles)(InfoDialog)