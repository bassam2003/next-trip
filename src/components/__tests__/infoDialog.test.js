import React from 'react'
import { shallow } from 'enzyme'
import { InfoDialog } from '../infoDialog'

describe('InfoDialog', () => {
  it('should render without error', () => {
    const wrapper = shallow(<InfoDialog />)

    expect(wrapper).toMatchSnapshot()
  })
})