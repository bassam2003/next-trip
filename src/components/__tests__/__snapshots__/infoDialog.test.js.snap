// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoDialog should render without error 1`] = `ShallowWrapper {}`;
